src/ocean/task/TaskPool.d

```diff
@@ -181,6 +185,9 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
 
         foreach (task; tasks_iterator)
         {
+            enforce(!isSame(this.toItem(current_task), this.toItem(task)),
```

Comment:
`assert` is not applicable here by our guidelines as there are no other safety checks against the situation and it will be removed in `-release`.